### DISCUSSION TITLE
TrueNAS + Home Assistant + securely remote access (NGINX or Traefik)

### DISCUSSION THREAD

**Discussion by megabri** [tprelog/iocage-homeassistant#43] [Q&A] [answered]
Dear All,
There is a sort of guide in order to configure HTTPS accesso to the Home Assistant Web Application in my TrueNAS ?

**Comment by tprelog** [comment]
I have not tried it myself, but maybe this is a good place to start

https://www.samueldowling.com/2020/01/18/nginx-reverse-proxy-freenas-ssl-tls/

**Comment by adorobis** [comment, marked as answer]
I have followed below guides and it works like a charm:
Set up NGINX to reverse proxy your jails w/ Certbot

<https://www.truenas.com/community/threads/how-to-set-up-nginx-to-reverse-proxy-your-jails-w-certbot.49876/> 


Quick and Dirty Dynamic DNS Using GoDaddy

<https://www.instructables.com/Quick-and-Dirty-Dynamic-DNS-Using-GoDaddy/> 

ACME protocol client written purely in Shell (Unix shell) language. - Automatic Certificate Management Environment 

<https://github.com/acmesh-official/acme.sh> 


deploy-freenas
deploy-freenas.py is a Python script to deploy TLS certificates to a FreeNAS/TrueNAS (Core)

<https://github.com/danb35/deploy-freenas>
